### PR TITLE
fix: correctly parse function argument types

### PR DESCRIPTION
**Pull request by jayeshp19**

### CODE REVIEW

File: tests/test_build_func_desc.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        name="foo",
+        description="foo",
+        auto_retry=False,
+        callable=lambda: None,
+        arguments={
+            "arg": FunctionArgInfo(
+                name="foo",
+                description="foo",
+                type=inner_type,
+                is_optional=is_optional,
```
Comment (theomonnom) [inline tests/test_build_func_desc.py:40]: type should be arg_typ directly for the test to be accurate?

**Comment by theomonnom** [comment]
let's make sure this work on python 3.9 before merging. I know the behaviours of Optional changed between python versions (With the new A | B syntax)

File: livekit-plugins/livekit-plugins-anthropic/livekit/plugins/anthropic/llm.py [hunk 2]
```diff
@@ -560,24 +562,26 @@ def type2str(t: type) -> str:
             raise ValueError(f"unsupported type {t} for ai_property")
 
         p: dict[str, Any] = {}
-        if arg_info.default is inspect.Parameter.empty:
+        if not _is_optional_type(arg_info.type)[0]:
```
Comment (theomonnom) [inline livekit-plugins/livekit-plugins-anthropic/livekit/plugins/anthropic/llm.py:565]: I think the logic is different?
Does it still accept default values?  `my_var: int = 5 `

Comment (theomonnom) [reply livekit-plugins/livekit-plugins-anthropic/livekit/plugins/anthropic/llm.py]: Can we add a test for it as well? (for default values)

Comment (jayeshp19) [reply livekit-plugins/livekit-plugins-anthropic/livekit/plugins/anthropic/llm.py]: we should count it as optional right?

Comment (theomonnom) [reply livekit-plugins/livekit-plugins-anthropic/livekit/plugins/anthropic/llm.py]: Yes, when sending it to OAI, they should not be inside the "`required`" field. Tho the type itself inside the Python code isn't really "optional"

File: tests/test_create_func.py [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+    assert arg_info.name == "param2"
+    assert arg_info.description == ""
+    assert arg_info.type is Optional[List[str]]
+    assert arg_info.default is None
+    assert arg_info.choices == ()
+    assert build_arg_info["type"] == "array"
+    assert build_arg_info["items"]["type"] == "string"
+
+    # check 'param3'
```
Comment (jayeshp19) [inline tests/test_create_func.py:119]: @theomonnom Added test case with default param

**Review by theomonnom — APPROVED** [review]

**Pull request merged**